docs/core/platform/application_credentials.md

```diff
@@ -0,0 +1,54 @@
+---
+title: "Application Credentials"
+---
+
+Integrations support [Configuration via OAuth2](https://developers.home-assistant.io/docs/config_entries_config_flow_handler#configuration-via-oauth2) and the preferred approach is to use the Home Assistant Cloud Account Linking service. Integrations may also allow users to provide their own OAuth client credentials by adding a `application_credentials.py` and implementing the right functions.
```

Comment:
The first sentence is missing something. Now it sounds like all integrations support oauth2.

Reply:
Also here, we should clarify the sentence about cloud account linking, as implementing a local implementation isn't optional.

Reply:
Apologies, I deleted the wrong thing, i'll get back to this after sleeping on it.

Reply:
OK, i revamped this to add more of the bigger picture overview. I also updated the config flow docs to point here, then added a link back here with details on cloud account linking and nabu casa.  Really, I think the cloud account linking stuff may deserve its own page.  Or we could move more of that detail here, with a dedicated section, even though it does not technically involve this platform it may be the natural place to show the progression.

Reply:
Yeah, a separate page about account linking may be needed. Although it's not so much for the contributor to do. It's mostly Nabu Casa that needs to set up an application. For now this is good, I think.